pgjdbc/src/main/java/org/postgresql/jdbc/TimestampUtils.java

```diff
@@ -988,9 +987,19 @@ public String toString(LocalDateTime localDateTime) {
         return "-infinity";
       }
 
-      // LocalDateTime is always passed with time zone so backend can decide between timestamp and timestamptz
-      ZonedDateTime zonedDateTime = localDateTime.atZone(getDefaultTz().toZoneId());
-      return toString(zonedDateTime.toOffsetDateTime());
+      sbuf.setLength(0);
+
+      if (nanosExceed499(localDateTime.getNano())) {
```

Comment:
Is this also a change from the current behaviour ?

Reply:
this should go to MIN/MAX PR

Reply:
This rounding was there, and it was done within `toString(ZonedDateTime)`